### PR TITLE
Add make target for sandbox gke clusters to simplify local dev/testing

### PR DESCRIPTION
**Pull request by haroldwoo**
Open for suggestions/comments/concerns

- Adds `make gke` and `make clean-gke` targets
~~Changes local environment to only load dags in the new devdags/ folder. (This may be controversial, but saves about 3-4 minutes of startup time involving importing dummy variables and connections)~~
- More info: go/wtmodev

infra pr https://github.com/mozilla-services/cloudops-infra/pull/2967

### CODE REVIEW

File: bin/start_gke
Comment (acmiyaguchi) [inline bin/start_gke:14]: You may want to set the project back at the end of the script, or use an explicit `--project` flag with the gcloud tool.

File: bin/run
Comment (acmiyaguchi) [inline bin/run:156]: I think we should make this configurable via an environment variable in the docker-compose file, so we can still get the old behavior by setting a flag.

File: docker-compose.yml
Comment (acmiyaguchi) [inline docker-compose.yml:25]: I'm not sure how ergonomic this will actually be. If I want to modify an existing workflow, I would have to eventually include the appropriate files and init variables. This is probably useful as standalone of the bigquery-etl workflow, though.

**Review by acmiyaguchi — COMMENTED** [review]
I think there should be a way to target the old behavior, even if it takes a few minutes to come online.

File: bin/run
Comment (fbertsch) [reply bin/run:156]: Agreed

**Review by fbertsch — COMMENTED** [review]
Echoing Anthony's comments on workflow. Most of the work we do in GKE Pod Operators involves some sort of connection/variable setup, simply because they don't fit in the bq_etl model.

File: docker-compose.yml
Comment (haroldwoo) [reply docker-compose.yml:25]: I'm not sure what you mean. If modifying an existing dag, you would need to move it to the devdags folder, what other files and variables are you referring to? IIRC the current workflow only injects dummy variables, so you would need to import the real variables when testing anyways.  In terms of files, I mount the operators/utils folders into the devdags folder as well.

Comment (haroldwoo) [reply docker-compose.yml:25]: It's also nice to only have to deal with the dag you're working on, and not load the others + vars/conns

Comment (acmiyaguchi) [reply docker-compose.yml:25]: Example: if I want to test `dags/mozaggregator_prerelease.py`, I would have to copy it over to the devdags directory to test, then copy it back over. You're right about the variables because I do already have import scripts handy.

File: docker-compose.yml
Comment (acmiyaguchi) [inline docker-compose.yml:49]: This breaks the `test-parse` CI, since the devdags folder is empty. We do want to test against all of the existing dags to make sure we're not making bad commits.

File: docker-compose.yml
Comment (haroldwoo) [reply docker-compose.yml:25]: Yeah this discussion is why I added you and fbertsch for review. 

IMO longer startup times, less logs, and little UI quirks (e.g. when toggling a dag on, and quickly refreshing, sometimes the wrong dag seems to be toggled on) would go away when only dealing with a single dag. 

The trade off is having to move dags before merging. But at least every new dag wouldn't require devs to add dummy vars/conns.

I did some testing and with prebuilt docker images, the startup time of local airflow without dags/conns/vars is ~10 seconds versus 3 minutes

File: bin/start_gke
Comment (whd) [inline bin/start_gke:7]: For bash style, we generally prefer `$()` to ``` `` ```. There appears to be only one use of ``` `` ``` in the top-level `airflow.sh` file (side note: looks like this can probably be deleted since it appears to reference AWS).

As to using an arbitrary site to look up your public ip... I'd really prefer if we had a better system than this, but anything else (e.g. IAP and GKE, bastion hosts) are probably significantly more effort. I'd almost prefer making the cluster accessible from anywhere since there's additional auth at the k8s layer.

File: bin/start_gke
Comment (whd) [inline bin/start_gke:52]:
```suggestion
```

File: bin/start_gke
Comment (whd) [inline bin/start_gke:51]:
```suggestion
echo "visit https://go.corp.mozilla.com/wtmodev for more info"
```

File: bin/start_gke
Comment (whd) [inline bin/start_gke:33]: Looks like I missed this in the ops PR. Is there a reason we don't allow `roles/iam.serviceAccountKeyAdmin` and allow created keys directly instead of using a static secret? In theory we could create / destroy these at the same cadence as the GKE clusters themselves, but we need to consider cleanup steps as well I suppose.

File: bin/start_gke
Comment (whd) [reply bin/start_gke:14]: I agree we should probably use `--project` explicitly instead of changing the user's general setting. Another option (that needs to be careful about error handling) is to get the current project before setting it to the gke and project and to reset the project after the script runs.

File: docker-compose.yml
Comment (whd) [reply docker-compose.yml:25]: I don't have any real opinion here (as I won't be a consumer of this particular tooling), but 10s seems a lot nicer than 3m. Ergonomically though copying things around doesn't seem ideal.

Comment (fbertsch) [reply docker-compose.yml:25]: Testing on this dev cluster is going to be a pretty high-touch piece anyways, so copying over is not really a big deal. I think we should be good to go with this improvement as long as we properly document this workflow in the README: https://github.com/mozilla/telemetry-airflow#testing-gke-jobs-including-bigquery-etl-changes

File: bin/start_gke
Comment (haroldwoo) [reply bin/start_gke]: just for simplicity. Also, isnt granting view on secrets better than granting serviceAccountKeyAdmin?

File: bin/start_gke
Comment (haroldwoo) [reply bin/start_gke:13]: theres another bug that causes the containers to restart locally, so i'll hold off on that

File: docker-compose.yml
Comment (haroldwoo) [reply docker-compose.yml]: rm'd.

File: docker-compose.yml
Comment (haroldwoo) [reply docker-compose.yml]: i've removed this piece and kept this pr involving gke only. If startup time is a goal in the future, we can revist this

File: bin/start_gke
Comment (haroldwoo) [reply bin/start_gke]: I've been using this site for years without issue. If we want to open this cluster to the public, it may widen the surface area of attack vectors as this is a public repo with the projectid and other things visible.

**Review by acmiyaguchi — APPROVED** [review]
I think this looks good as is. 

There are some good ideas with mounting the /devdags folder that I think may be useful elsewhere (particularly bigquery-etl), but we'll want to be careful how it's configured to ensure that we're running CI correctly on PRs.

File: bin/start_gke
Comment (whd) [reply bin/start_gke]: Since this is a development environment tool it's probably fine to rely on such a third-party service, but I'd request using `$(curl https://icanhazip.com)` and  a comment/link to https://major.io/icanhazip-com-faq/#what-about-my-privacy, as well as the option to override this via environment variable (i.e. only run this when the variable isn't set).

Again though, I'm not likely to use this tooling so if others are fine with it as-is I'm not going to object.

File: bin/start_gke
Comment (whd) [reply bin/start_gke]: The `serviceAccountKeyAdmin` idiom is modeled after cloudops-deploylib and has the advantage that secrets are temporal (they're created per-run and deleted afterward). It requires proper cleanup though and we have an [open issue](https://github.com/mozilla-services/cloudops-infra-deploylib/issues/105) for that in deploylib land.

On the whole I'd prefer to not have static secrets that are susceptible to requiring rotation (when e.g. a DE leaves) but I'm also not strenuously objecting to it, especially when we haven't always had success implementing the temporal idiom in more secure environments.

**Pull request merged**
